homeassistant/components/intent_script/__init__.py

```diff
@@ -99,11 +99,13 @@ async def async_handle(self, intent_obj):
                 speech[CONF_TYPE],
             )
 
-        if reprompt is not None and reprompt[CONF_TEXT].template:
-            response.async_set_reprompt(
-                reprompt[CONF_TEXT].async_render(slots, parse_result=False),
-                reprompt[CONF_TYPE],
-            )
+        if reprompt is not None:
+            text_reprompt = reprompt[CONF_TEXT].async_render(slots, parse_result=False)
+            if text_reprompt:
+                response.async_set_reprompt(
+                    text_reprompt,
+                    reprompt[CONF_TYPE],
+                )
```

Comment:
How is this change tested?

Reply:
Currently, I don't know how to create a test case from the very beginning. I only know how to modify current ones.
I understand how alexa intents works, but I don't know the whole hass lifecycle to be able to create a test case.
If you help me with 1 example for this case, I can replicate to the other ones this weekend.

Reply:
I think you can just add a new test to `tests/components/intent_script/test_init.py` which fails without the change in this PR and passes with it.

In the test, set reprompt so it is or renders as empty, and then check that the remprompt is not present in the response.

Reply:
I think it is pacified that code won't need to change, as I already added a test as requested.
Unless you have any other observation.

Do you consider it resolved too?